### DISCUSSION TITLE
Extended operating system support

### DISCUSSION THREAD

**Discussion by rashil2000** [lptstr/winfetch#53] [Ideas]
For [Project 5](https://github.com/lptstr/winfetch/projects/5) and [Project 6](https://github.com/lptstr/winfetch/projects/6), here are the logos:

``` 
++++++++++++++++++++hMN/..............
+++++++++++++++++++dMm-...............
++++++++++++++++++dMd-................
++++++++mms++++++dMd-.....ydo.........
++++++++MMs+++++yMN-......dMs.........
++++++++NNs++++oNM/.......hNs.........
+++++++++++++++dMh--------------------
oooooooooooooooMM/--------------------
oooooooooooooohMm---------------------
oooooooooooooomMy---------------------
ooooooooooooooNMs---------------------
ssssssssssssssNMMMMMMMMo:::::::::::::-
ssssssssssssssyyyyyyhMM:::::::::::::::
sssshhyssssssssssssshMM::::::::/oo::::
sssymMMMNmmdhhyyyyyyhMM+osyhdmMMNd/:::
ssssssyhhdmmNNMMMMMMMMMMNmdhyso/::::::
ssssssssssssssssssyyyNMy::::::::::::::
sssssssssssssssssssssdMN::::::::::::::
ssssssssssssssssssssssNMs:::::::::::::
```

```
                .88888888:.
               88888888.88888.
             .8888888888888888.
             888888888888888888
             88' _`88'_  `88888
             88 88 88 88  88888
             88_88_::_88_:88888
             88:::,::,:::::8888
             88`:::::::::'`8888
            .88  `::::'    8:88.
           8888            `8:888.
         .8888'             `888888.
        .8888:..  .::.  ...:'8888888:.
       .8888.'     :'     `'::`88:88888
      .8888        '         `.888:8888.
     888:8         .           888:88888
   .888:88        .:           888:88888:
   8888888.       ::           88:888888
   `.::.888.      ::          .88888888
  .::::::.888.    ::         :::`8888'.:.
 ::::::::::.888   '         .::::::::::::
 ::::::::::::.8    '      .:8::::::::::::.
.::::::::::::::.        .:888:::::::::::::
:::::::::::::::88:.__..:88888:::::::::::'
 `'.:::::::::::88888888888.88:::::::::'
       `':::_:' -- '' -'-' `':_::::'`
```

**Comment by kiedtl** [comment]
Honestly, I think this is a pretty terrible idea.

- While on Windows PowerShell has the blessing of MSFT, is the de-facto scripting language for sysadmins, and is installed by default on every OS newer than Windows 7, the vast majority of those on other operating systems don't use PowerShell (and most just hate it) and won't have it installed.
- There are already well-tested mature information fetching tools (Neofetch, pfetch, and friends).

Adding support for other non-Windows oses would require adding a ton of new code and increase the maintainence burden, and I don't see very many people using it anyway. I don't think its worth it.

Feel free to yell at me if I'm not making any sense :^)

**Reply by jcwillox** [reply]
This is exactly what I was thinking, without `Get-CimInstance` we would have to add custom methods to get info on every different OS, seriously complicating winfetch. Neofetch has a much larger community around it and it would take considerable work to create anything that could rival it in PowerShell. It'd be much better to capitalise on the market where neofetch doesn't work well, which is Windows. If we were creating a more efficient, proper cross-platform tool like a `rust` or `go` based version, it'd make sense, I doubt people will want to install PowerShell just for winfetch.

**Reply by rashil2000** [reply]
People obviously won't install PowerShell just for winfetch; the idea was for those who already use it there (like myself).

I was initially a bit excited for winfetch so I decided to make up Project 5 and 6. But like both of you, even I have grown wary of actually implementing this - albeit for a different reason. And that reason is: Powershell is an extremely modern shell/language. While traditional shells like CMD/Bash/Zsh rely on program binaries for literally everything, in PowerShell, each command is a shell builtin (Bash and CMD also have shell builtins, but they're like 50 in total as compared to PowerShell's almost 2000). The obvious advantage of this is that PowerShell can function fully well even if your PATH variable is completely empty; most other shells get crippled if `/usr/bin` or `C:\Windows\System32` is removed from PATH. I agree this is an extremely unlikely scenario, but shell builtins are usually faster as compared to calling binaries (binaries get executed as a separate process). 
And on macOS/Linux, we would have to call a ton of binaries, apart from doing repeated `Get-Content`'s to read files (most information in Linux is stored in files; I think people call this as 'everything is a file' principle, correct me if I'm wrong) to get system information, something that I personally dislike. 

But yes, the feeling is mutual 😅
